### ISSUE TITLE
IN operator with GUID fails

### ISSUE THREAD

**Issue by Casimodo72** [OData/WebApi#1596]
An HTTP OData V4 request using a filter with the IN operator with GUIDs fails.

Strangely the request works (but shouldn't) when using single quotation marks around the GUIDs.

(Side note: is there a broader data type issue w.r.t the IN operator? Because I can't make the IN operator to work with integers at all).

### Assemblies affected

OData WebApi lib 7.0.1

### Reproduce steps

HTTP request. This fails but should not fail:
...  /Foo?$filter=Id in (27811919BD-323C-44B0-A7FD-0081B730F829)

This works (note the additional quotation marks) but it shouldn't work:
...  /Foo?$filter=Id in ('27811919BD-323C-44B0-A7FD-0081B730F829')

### Expected result

Expecting no errors when using GUID as defined by the OData V4 URL convention spec; i.e. without quotation marks.

### Actual result

I'm getting an error:

Message: "The query specified in the URI is not valid. Invalid JSON. A comma character ',' was expected in scope 'Array'. Every two elements in an array and properties of an object must be separated by commas."

Exception type: "Microsoft.OData.ODataException"

### Additional detail
Using
Microsoft.OData.Core 7.5.0
.NET 4.6

**Comment by KanishManuja-MS** [comment]
@Casimodo72 This does look like a bug. I am actively working on a fix for this.

**Comment by KanishManuja-MS** [comment]
We determined that this issue is on the OData side as the JSON reader is not able to read edm primitive types in it's raw format.

**Comment by zamzowd** [comment]
While it looks like there was a fix for this released in 7.5.3, I'm still having issues when using `in` inside of an `any` operator, e.g. `$filter=customers/any(c:c/id in (<guid1>,<guid2>))`.  Instead of a 400 Bad Request, I get a server error.  The property of the related collection is not nullable.

Here is the exception logged by the server:
```
"typeName": "System.ArgumentException",
"message": "Expression of type 'System.Collections.Generic.List`1[System.Nullable`1[System.Guid]]' cannot be used for parameter of type 'System.Collections.Generic.IEnumerable`1[System.Guid]' of method 'Boolean Contains[Guid](System.Collections.Generic.IEnumerable`1[System.Guid], System.Guid)'\r\nParameter name: arg0",
"hasFullStack": true,
"parsedStack": [
{
  "level": 0,
  "method": "System.Dynamic.Utils.ExpressionUtils.ValidateOneArgument",
  "assembly": "System.Linq.Expressions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
  "line": 0
},
{
  "level": 1,
  "method": "System.Linq.Expressions.Expression.Call",
  "assembly": "System.Linq.Expressions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
  "line": 0
},
{
  "level": 2,
  "method": "Microsoft.AspNet.OData.Query.Expressions.FilterBinder.BindAnyNode",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 3,
  "method": "Microsoft.AspNet.OData.Query.Expressions.FilterBinder.BindExpression",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 4,
  "method": "Microsoft.AspNet.OData.Query.Expressions.FilterBinder.BindFilterClause",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 5,
  "method": "Microsoft.AspNet.OData.Query.FilterQueryOption.ApplyTo",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 6,
  "method": "Microsoft.AspNet.OData.Query.ODataQueryOptions.ApplyTo",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 7,
  "method": "Microsoft.AspNet.OData.EnableQueryAttribute.ExecuteQuery",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 8,
  "method": "Microsoft.AspNet.OData.EnableQueryAttribute.OnActionExecuted",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 9,
  "method": "Microsoft.AspNet.OData.EnableQueryAttribute.OnActionExecuted",
  "assembly": "Microsoft.AspNetCore.OData, Version=7.1.0.21120, Culture=neutral, PublicKeyToken=31bf3856ad364e35",
  "line": 0
},
{
  "level": 10,
  "method": "Microsoft.AspNetCore.Mvc.Filters.ActionFilterAttribute+<OnActionExecutionAsync>d__6.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 11,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 12,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 13,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+<InvokeNextActionFilterAsync>d__10.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 14,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 15,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 16,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 17,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+<InvokeInnerFilterAsync>d__13.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 18,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 19,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 20,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeNextResourceFilter>d__23.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 21,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 22,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 23,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 24,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeFilterPipelineAsync>d__18.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 25,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 26,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 27,
  "method": "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+<InvokeAsync>d__16.MoveNext",
  "assembly": "Microsoft.AspNetCore.Mvc.Core, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 28,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 29,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 30,
  "method": "Microsoft.AspNetCore.Builder.RouterMiddleware+<Invoke>d__4.MoveNext",
  "assembly": "Microsoft.AspNetCore.Routing, Version=2.2.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 31,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 32,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 33,
  "method": "Microsoft.AspNetCore.Authentication.AuthenticationMiddleware+<Invoke>d__6.MoveNext",
  "assembly": "Microsoft.AspNetCore.Authentication, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 34,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 35,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 36,
  "method": "Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware+<Invoke>d__7.MoveNext",
  "assembly": "Microsoft.AspNetCore.StaticFiles, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
},
{
  "level": 37,
  "method": "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 38,
  "method": "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification",
  "assembly": "System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
  "line": 0
},
{
  "level": 39,
  "method": "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware+<Invoke>d__7.MoveNext",
  "assembly": "Microsoft.AspNetCore.Diagnostics, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
  "line": 0
}
]
```

**Comment by gathogojr** [comment]
Guids not enclosed in single quotes supported in current version: Microsoft.AspNetCore.OData 7.5.14
